tools/shared.py

```diff
@@ -1714,18 +1714,21 @@ def is_ar(filename):
 
   @staticmethod
   def is_bitcode(filename):
-    # look for magic signature
-    b = open(filename, 'r').read(4)
-    if b[0] == 'B' and b[1] == 'C':
-      return True
-    # look for ar signature
-    elif Building.is_ar(filename):
-      return True
-    # on OS X, there is a 20-byte prefix
-    elif ord(b[0]) == 222 and ord(b[1]) == 192 and ord(b[2]) == 23 and ord(b[3]) == 11:
-      b = open(filename, 'r').read(24)
-      return b[20] == 'B' and b[21] == 'C'
-
+    try:
+        # look for magic signature
+        b = open(filename, 'r').read(4)
+        if b[0] == 'B' and b[1] == 'C':
+          return True
+        # look for ar signature
+        elif Building.is_ar(filename):
+          return True
+        # on OS X, there is a 20-byte prefix
+        elif ord(b[0]) == 222 and ord(b[1]) == 192 and ord(b[2]) == 23 and ord(b[3]) == 11:
+          b = open(filename, 'r').read(24)
+          return b[20] == 'B' and b[21] == 'C'
+    except IndexError:
+        # The caller function should display a error in the log
```

Comment:
I don't think there is a need for this log because the caller function already does this. We can see this at: 

emcc.py, line 809
>  if shared.Building.is_bitcode(arg):
>               logging.error(\***omitted log message**\*)
> else:
>               logging.error(arg + ': Unknown format, not a static library!')
>             exit(1)

and so I'd say that adding more logs inside this function would make the log file a little unreadable
Btw, I'll make another PR